python/pyarrow/tests/test_dataset.py

```diff
@@ -86,6 +86,19 @@ def _table_from_pandas(df):
     return table.replace_schema_metadata()
 
 
+def assert_dataset_fragment_convenience_methods(dataset):
+    # FileFragment convenience methods
+    for fragment in dataset.get_fragments():
+        if pq is not None and isinstance(dataset.format, ds.ParquetFileFormat):
+            assert isinstance(fragment.metadata, pq.FileMetaData)
```

Comment:
Since the `metadata` for Parquet is not really related to the open() method, I would move that to a separate test (we should have some Parquet fragment specific tests, I see a bunch that start with `test_fragments_parquet_...`)